crates/pop-cli/src/common/wallet.rs

```diff
@@ -65,3 +69,30 @@ pub fn prompt_to_use_wallet(cli: &mut impl Cli) -> anyhow::Result<bool> {
 		Ok(false)
 	}
 }
+
+// Sign and submit an extrinsic using wallet integration, then returns the resulting events.
+#[cfg(feature = "parachain")]
+pub(crate) async fn submit_extrinsic_with_wallet(
+	client: &OnlineClient<SubstrateConfig>,
+	url: &Url,
+	call_data: Vec<u8>,
+	cli: &mut impl Cli,
+) -> Result<ExtrinsicEvents<SubstrateConfig>> {
+	let maybe_payload = request_signature(call_data, url.to_string()).await?;
+	if let Some(payload) = maybe_payload {
```

Comment:
```suggestion
	let payload = maybe_payload.map_err(|_| Err(anyhow!("No signed payload received.")))?;
```

Another minor improvement, by mapping the error here, we can avoid the `if-esle` statement.

Reply:
Nice. Refactored https://github.com/r0gue-io/pop-cli/pull/410/commits/3b9aace4830ce4afc4f60d6c5dccf65af63cf9e4